castai/resource_castai_cluster.go

```diff
@@ -217,7 +399,101 @@ func waitForClusterToReachStatusFunc(ctx context.Context, client *sdk.ClientWith
 				return resource.RetryableError(fmt.Errorf("waiting for cluster to reach %q status, id=%q name=%q, status=%s", targetStatus, cluster.Id, cluster.Name, cluster.Status))
 			}
 		}
-
 		return resource.NonRetryableError(fmt.Errorf("cluster has reached unexpected status, id=%q name=%q, status=%s", cluster.Id, cluster.Name, cluster.Status))
 	}
 }
+
+func expandCPUlimit(params interface{}) sdk.ClusterLimitsCpu {
+	var clusterLimitsCPU sdk.ClusterLimitsCpu
+	for _, val := range params.([]interface{}) {
+		cpuData := val.(map[string]interface{})
+		clusterLimitsCPU = sdk.ClusterLimitsCpu{
+			MaxCores:  int64(cpuData["max_cores"].(int)),
+			MinCores:  int64(cpuData["min_cores"].(int)),
+		}
+	}
+	return clusterLimitsCPU
+}
+
+func expandNodeDownscaler(params interface{}) *sdk.NodeDownscalerEmptyNodes {
+	var nodeDownscalerBool sdk.NodeDownscalerEmptyNodes
```

Comment:
Either rename this to expandNodeDownscalerEmptyNodes or better:
```	for _, val := range pc[PolicyFieldNodeDownscaler].([]interface{}) {
		ndData := val.(map[string]interface{})
		nodeDownscaler = sdk.NodeDownscaler{
			EmptyNodes: expandNodeDownscaler(ndData["empty_nodes"].(interface{})),
		}
	}```

extract the `sdk.NodeDownscaler` part and put it into this expansion func.

Reply:
adjusted particular place, should I do it for all policy blocks?